src/einsteinpy/symbolic/tensor.py

```diff
@@ -266,9 +266,16 @@ def __init__(
             Raised when arguments syms, variables, functions have data type other than list, tuple or set.
         TypeError
             Raised when argument parent_metric does not belong to MetricTensor class and isn't None.
+        ValueError
+            Raised when argument syms does not agree with shape of argument arr
 
         """
         super(BaseRelativityTensor, self).__init__(arr=arr, config=config, name=name)
+
+        if len(self.arr.shape) != 0:
+            if self.arr.shape[0] != len(syms):
+                raise ValueError(f"invalid shape of array for syms: {syms}")
```

Comment:
I guess you are worried about `IndexError` if you use `and` between the 2 conditions, but it won't happen.
Replace with this, we don't want unnecessary cyclometric complexity.

```
if len(self.arr.shape) != 0 and self.arr.shape[0] != len(syms):
```